### PR TITLE
[Merged by Bors] - feat: port #18526

### PR DESCRIPTION
**Pull request by vihdzp**
Mathlib 3: https://github.com/leanprover-community/mathlib/pull/18526/files

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Order/Antisymmetrization.lean
Comment (eric-wieser) [inline Mathlib/Order/Antisymmetrization.lean:1]: It looks like you didn't forward-port the golfs to `ofAntisymmetrization_le_ofAntisymmetrization_iff` and `ofAntisymmetrization_lt_ofAntisymmetrization_iff `

File: Mathlib/Order/RelIso/Basic.lean [hunk 1]
```diff
@@ -393,33 +393,82 @@ protected theorem isWellOrder : ∀ (_ : r ↪r s) [IsWellOrder β s], IsWellOrd
...
-noncomputable def _root_.Quotient.outRelEmbedding [s : Setoid α] {r : α → α → Prop}
+noncomputable def Quotient.outRelEmbedding [Setoid α] {r : α → α → Prop}
     (H : ∀ (a₁ b₁ a₂ b₂ : α), a₁ ≈ a₂ → b₁ ≈ b₂ → r a₁ b₁ = r a₂ b₂) : Quotient.lift₂ r H ↪r r :=
   ⟨Embedding.quotientOut α, by
     refine' @fun x y => Quotient.inductionOn₂ x y fun a b => _
     apply iff_iff_eq.2 (H _ _ _ _ _ _) <;> apply Quotient.mk_out⟩
 #align quotient.out_rel_embedding Quotient.outRelEmbedding
 #align quotient.out_rel_embedding_apply Quotient.outRelEmbedding_apply
 
+/-- `quotient.out'` as a relation embedding between the lift of a relation and the relation. -/
```
Comment (eric-wieser) [inline Mathlib/Order/RelIso/Basic.lean:415]:
```suggestion
/-- `Quotient.out'` as a relation embedding between the lift of a relation and the relation. -/
```

File: Mathlib/Order/RelIso/Basic.lean [hunk 2]
```diff
@@ -393,33 +393,82 @@ protected theorem isWellOrder : ∀ (_ : r ↪r s) [IsWellOrder β s], IsWellOrd
...
+    (H : ∀ (a₁ b₁ a₂ b₂ : α), a₁ ≈ a₂ → b₁ ≈ b₂ → r a₁ b₁ = r a₂ b₂) :
+    (fun a b => Quotient.liftOn₂' a b r H) ↪r r :=
+  { Quotient.outRelEmbedding _ with toFun := Quotient.out' }
+#align rel_embedding.quotient.out'_rel_embedding Quotient.out'RelEmbedding
+
+@[simp]
+theorem acc_lift₂_iff [Setoid α] {r : α → α → Prop}
+    {H : ∀ (a₁ b₁ a₂ b₂ : α), a₁ ≈ a₂ → b₁ ≈ b₂ → r a₁ b₁ = r a₂ b₂} {a} :
+    Acc (Quotient.lift₂ r H) ⟦a⟧ ↔ Acc r a :=
+  by
```
Comment (eric-wieser) [inline Mathlib/Order/RelIso/Basic.lean:427]:
```suggestion
    Acc (Quotient.lift₂ r H) ⟦a⟧ ↔ Acc r a := by
```

File: Mathlib/Order/RelIso/Basic.lean [hunk 3]
```diff
@@ -393,33 +393,82 @@ protected theorem isWellOrder : ∀ (_ : r ↪r s) [IsWellOrder β s], IsWellOrd
...
+@[simp]
+theorem acc_liftOn₂'_iff {s : Setoid α} {r : α → α → Prop} {H} {a} :
+    Acc (fun x y => Quotient.liftOn₂' x y r H) (Quotient.mk'' a : Quotient s) ↔ Acc r a :=
+  acc_lift₂_iff
+#align acc_lift_on₂'_iff acc_liftOn₂'_iff
+
 /-- A relation is well founded iff its lift to a quotient is. -/
 @[simp]
-theorem _root_.wellFounded_lift₂_iff [s : Setoid α] {r : α → α → Prop}
+theorem wellFounded_lift₂_iff [s : Setoid α] {r : α → α → Prop}
```
Comment (eric-wieser) [inline Mathlib/Order/RelIso/Basic.lean:446]:
```suggestion
theorem wellFounded_lift₂_iff [Setoid α] {r : α → α → Prop}
```
according to the dashboard

File: Mathlib/Order/RelIso/Basic.lean [hunk 4]
```diff
@@ -393,33 +393,82 @@ protected theorem isWellOrder : ∀ (_ : r ↪r s) [IsWellOrder β s], IsWellOrd
...
 alias wellFounded_lift₂_iff ↔ _root_.WellFounded.of_quotient_lift₂ _root_.WellFounded.quotient_lift₂
 
+@[simp]
+theorem wellFounded_liftOn₂'_iff {s : Setoid α} {r : α → α → Prop} {H} :
+    (WellFounded fun x y : Quotient s => Quotient.liftOn₂' x y r H) ↔ WellFounded r :=
+  wellFounded_lift₂_iff
+#align well_founded_lift_on₂'_iff wellFounded_liftOn₂'_iff
+
+alias wellFounded_liftOn₂'_iff ↔
+  wellFounded.of_quotient_liftOn₂' wellFounded.quotient_liftOn₂'
```
Comment (eric-wieser) [inline Mathlib/Order/RelIso/Basic.lean:466]:
```suggestion
  WellFounded.of_quotient_liftOn₂' WellFounded.quotient_liftOn₂'
```

File: Mathlib/Order/RelIso/Basic.lean [hunk 5]
```diff
@@ -393,33 +393,82 @@ protected theorem isWellOrder : ∀ (_ : r ↪r s) [IsWellOrder β s], IsWellOrd
...
+#align well_founded.of_quotient_lift_on₂' wellFounded.of_quotient_liftOn₂'
+#align well_founded.quotient_lift_on₂' wellFounded.quotient_liftOn₂'
```
Comment (eric-wieser) [inline Mathlib/Order/RelIso/Basic.lean:468]: The mathport output here was correct, but it seems you edited it to be incorrect.

Comment (vihdzp) [reply Mathlib/Order/RelIso/Basic.lean:467]: Sorry, I don't quite get the capitalization rules. Is `WellFounded` always uppercase like that?

Comment (eric-wieser) [reply Mathlib/Order/RelIso/Basic.lean:467]: Well you're trying to put these lemmas in the namespace that matches the type, right? And the `wellFounded` namespace doesn't exist.

File: Mathlib/Order/Antisymmetrization.lean
Comment (vihdzp) [reply Mathlib/Order/Antisymmetrization.lean:1]: Whoever originally ported that file had already golfed them. I'd be increasing the proof length if I forward-ported the Lean 3 golfs.

Comment (eric-wieser) [reply Mathlib/Order/Antisymmetrization.lean:1]: I don't agree. See the extra commit I pushed.

File: Mathlib/Order/RelIso/Basic.lean [hunk 6]
```diff
@@ -393,32 +393,80 @@ protected theorem isWellOrder : ∀ (_ : r ↪r s) [IsWellOrder β s], IsWellOrd
   | f, H => { f.isStrictTotalOrder with wf := f.wellFounded H.wf }
 #align rel_embedding.is_well_order RelEmbedding.isWellOrder
 
+end RelEmbedding
```
Comment (eric-wieser) [inline Mathlib/Order/RelIso/Basic.lean:396]: Where did this come from? I don't see it in the mathport diff.

Comment (vihdzp) [reply Mathlib/Order/RelIso/Basic.lean:396]: I added it to avoid writing `_root_` a bunch of times.

Comment (eric-wieser) [reply Mathlib/Order/RelIso/Basic.lean:396]: I think it's best to just port faithfully; the more you refactor when porting the harder it becomes to forward-port subsequent changes.

**Comment by semorrison** [comment]
@vihdzp, it's helpful also if you include the link to the porting dashboard page.

I find this link by clicking on #port-status from zulip (or bookmarking https://leanprover-community.github.io/mathlib-port-status/), clicking on out-of-sync, finding the relevant file (which ideally would be mentioned in the PR title), and then copying and pasting the markdown code for the link that you see at the bottom of the left column:

* [`order.antisymmetrization`@`f1a2caaf51ef593799107fe9a8d5e411599f3996`..`3353f661228bd27f632c600cd1a58b874d847c90`](https://leanprover-community.github.io/mathlib-port-status/file/order/antisymmetrization?range=f1a2caaf51ef593799107fe9a8d5e411599f3996..3353f661228bd27f632c600cd1a58b874d847c90)

Doing this on forward porting PRs makes life easier for reviewers, thanks!

**Comment by semorrison** [comment]
bors d+

and please review the requests above, for future PRs, thanks!

**Pull request closed without merging**